src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -647,28 +666,50 @@ public void calculateContingencySensitivityValues(PropagatedContingency continge
                                                       Collection<LfBus> disabledBuses, Collection<LfBranch> disabledBranches, Reporter reporter) {
         List<LfSensitivityFactor<DcVariableType, DcEquationType>> factors = factorHolder.getFactorsForContingency(contingency.getContingency().getId());
-        if (contingency.getHvdcIdsToOpen().isEmpty() && contingency.getGeneratorIdsToLose().isEmpty()) {
+        if (contingency.getHvdcIdsToOpen().isEmpty() && contingency.getGeneratorIdsToLose().isEmpty() && contingency.getLoadIdsToLose().isEmpty()) {
             calculateSensitivityValues(factors, factorStates, contingenciesStates, flowStates, contingencyElements,
                     contingency, valueWriter);
         } else {
-            // if we have a contingency including the loss of a DC line or a generator
-            // if we have a contingency including the loss of a generator
+            // if we have a contingency including the loss of a DC line or a generator or a load
 
             Map<LfBus, BusState> busStates = new HashMap<>();
             Set<LfGenerator> participatingGeneratorsToRemove = new HashSet<>();
-            applyInjectionContingencies(network, lfNetwork, contingency, participatingGeneratorsToRemove, busStates, lfParameters);
+            Set<Load> participatingLoadsToRemove = new HashSet<>();
+            applyInjectionContingencies(network, lfNetwork, contingency, participatingGeneratorsToRemove, participatingLoadsToRemove, busStates, lfParameters);
 
             List<ParticipatingElement> newParticipatingElements = participatingElements;
             DenseMatrix newFactorStates = factorStates;
             boolean participatingElementsChanged = !participatingGeneratorsToRemove.isEmpty()
+                || !participatingLoadsToRemove.isEmpty()
                 || (isDistributedSlackOnGenerators(lfParameters) && !contingency.getGeneratorIdsToLose().isEmpty())
-                || (isDistributedSlackOnLoads(lfParameters) && !contingency.getHvdcIdsToOpen().isEmpty());
+                || (isDistributedSlackOnLoads(lfParameters) && (!contingency.getHvdcIdsToOpen().isEmpty() || !contingency.getLoadIdsToLose().isEmpty()));
             if (participatingElementsChanged) {
                 // deep copy of participatingElements, removing the participating LfGeneratorImpl whose targetP has been set to 0
                 newParticipatingElements = participatingElements.stream()
                     .filter(participatingElement -> !participatingGeneratorsToRemove.contains(participatingElement.getElement()))
                     .map(participatingElement -> new ParticipatingElement(participatingElement.getElement(), participatingElement.getFactor()))
                     .collect(Collectors.toList());
+                // modify newParticipatingElements, decreasing participation factors of lfBuses losing some loads
```

Comment:
Here I think that you have to do what have be done for generators and the `newParticipatingElements` building. If your bus that has lost a load has the updated loadTargetP, it will be automatic.